### PR TITLE
fix internal error when using rustfmt::skip with newline on stmt

### PR DESCRIPTION
**Pull request by rchaser53**
fix: #3717
This PR is to fix the bug reported by BartMassey. I will send the another PR to fix other case including 0xpr03's bug.

### CODE REVIEW

File: src/visitor.rs [hunk 1]
```diff
@@ -120,11 +120,14 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
...
                 if contains_skip(attrs) {
+                    let span = stmt.span();
                     self.push_skipped_with_span(
                         attrs,
-                        stmt.span(),
+                        span,
                         get_span_without_attrs(stmt.as_ast_node()),
                     );
+                    let context = self.get_context();
+                    add_skipped_range(&context, span);
```
Comment (topecongiro) [inline src/visitor.rs]: `push_skipped_with_span` is super tricky. Even though we skip formatting any statement/item/expression with `rustfmt::skip` attribute, we do **not** want to skip formatting the attribute itself. E.g.,

```rust
#[rustfmt::skip]  

let x = 3  ;
```

in the above example, even though we do not modify `let x = 3  ;`, we do want to trim trailing spaces after `#[rustfmt::skip]` itself and any empty lines between the attribute and the let statement (though I don't think we do this correctly in every case).

Comment (topecongiro) [reply src/visitor.rs]: The takeaway here is that we do not want to call `add_skipped_range` with a span which includes `#[rustfmt::skip]`.

**Comment by rchaser53** [comment]
I misunderstood. We only need the diffs in src/expr.rs to fix the issue. Sorry!

File: src/expr.rs [hunk 2]
```diff
@@ -517,6 +517,11 @@ pub(crate) fn rewrite_block_with_visitor(
     let inner_attrs = attrs.map(inner_attributes);
     let label_str = rewrite_label(label);
     visitor.visit_block(block, inner_attrs.as_ref().map(|a| &**a), has_braces);
+    let visitor_context = visitor.get_context();
+    context
+        .skipped_range
+        .borrow_mut()
+        .append(&mut visitor_context.skipped_range.borrow_mut().clone());
```
Comment (topecongiro) [inline src/expr.rs]: Nit: we can consume `visitor_context` and avoid `clone`ing here, though I assume that requires an additional refactoring.

Comment (rchaser53) [reply src/expr.rs]: Is it better to delete the clone than to use replace_with? The following code can also be compiled:

```
context
     .skipped_range
     .borrow_mut()
     .append(&mut visitor_context.skipped_range.borrow_mut());
```

Comment (rchaser53) [reply src/expr.rs]: I think just remove clone is better than using replace_with.

Comment (topecongiro) [reply src/expr.rs]: Great, this is much simpler than I first expected :smile: Thanks!

**Pull request merged**